.github/workflows/docker.yaml

```diff
@@ -0,0 +1,52 @@
+name: Docker
+on:
+  pull_request:
+    branches:
+    - main
```

Comment:
Btw, I know this was originally on `push` , but since its running really fast, is it ok to just have this run on every `pull_request` to main? @dblalock

Reply:
will that result in every PR overwriting the image on dockerhub? slash, could someone just open some random PR that corrupts the image and have it distributed to everyone who docker pulls it?

Reply:
After discussing from Bandish, my understanding of what will happen now is:
* inside PR, every commit will **build** but **not push** a new image. So you can check with Github Action CI if the image builds or breaks. There is not a direct way to try running on the image since it is not pushed
* On PR approval and merge,  it will **rebuild** and **push** to `mosaicml/examples:[llm]-[main-commithash]` and `mosaicml/exmaples:[llm]-[latest]`. 

So we will not overwrite any tags that have a hash. And we will only overwrite the `-latest` ones.